### PR TITLE
LL-7348 - SWAP - Test useProviderRates hook 

### PR DESCRIPTION
**Pull request by jelbaz-ledger**
## Context (issues, jira)

[LL-7348]

## Expectations

- [x] **Test coverage: The changes of this PR are covered by test.** Unit test were added with mocks when depending on a backend/device.
- [x] **No impact: The changes of this PR have ZERO impact on the userland.** Meaning, we can use these changes without modifying LLD/LLM at all. It will be a "noop" and the maintainers will be able to bump it without changing anything.

[LL-7348]: https://ledgerhq.atlassian.net/browse/LL-7348

### CODE REVIEW

File: src/exchange/swap/getExchangeRates.ts [hunk 1]
```diff
@@ -16,16 +16,19 @@ import {
 const getExchangeRates: GetExchangeRates = async (
   exchange: Exchange,
   transaction: Transaction,
-  userId?: string // TODO remove when wyre doesn't require this for rates
+  userId?: string, // TODO remove when wyre doesn't require this for rates
```
Comment (juan-cortes) [inline src/exchange/swap/getExchangeRates.ts:19]: this is no longer required for Wyre to give us the rates, you can verify (connected to a US vpn)
```
curl -L -X POST 'https://swap.ledger.com/v3/rate' \
-H 'Content-Type: application/json' \
--data-raw '{"from": "bitcoin", "to": "ethereum", "amountFrom": "1"}'
```

File: src/exchange/swap/getExchangeRates.ts [hunk 2]
```diff
@@ -16,16 +16,19 @@ import {
...
+  currencyTo?: TokenCurrency | CryptoCurrency | undefined | null
```
Comment (juan-cortes) [inline src/exchange/swap/getExchangeRates.ts:20]: yikes

Comment (jelbaz-ledger) [reply src/exchange/swap/getExchangeRates.ts:20]: Yeah this is not ideal but I tried to keep it compatible with the old swap :/. 
I guess the alternative would have been to copy/paste the function and change the args which is not super nice either.

File: src/exchange/swap/getExchangeRates.ts (hunk 1 above)
Comment (jelbaz-ledger) [reply src/exchange/swap/getExchangeRates.ts:19]: I'll double check 👍

File: src/exchange/swap/getExchangeRates.ts (hunk 2 above)
Comment (juan-cortes) [reply src/exchange/swap/getExchangeRates.ts:20]: nah, this is good, it's just ugly 🤣

**Review by juan-cortes — APPROVED** [review]

**Review by imqdee — APPROVED** [review]

**Pull request merged**